Record app export feature

jrchudy opened the pull request:
This PR adds export to `record` app. This adds a default download as CSV option that will always show up on `record` app now in the dropdown. If there is an existing export annotation for the table represented by the record page, the options defined by the annotation will show up in the dropdown menu.

A minor change also included was the use of `allowAnimations(false)` which can be called on any element found using protractor (angular) selectors. This removes the fade animation for when the modal is closed so that a click event that occurs in the next test case doesn't try to click on the modal or it's backdrop while it's fading out.

RFSH commented on record/index.html.in:
```diff
@@ -39,6 +39,9 @@
                     </strong>
                 </a>&nbsp;&nbsp;
+                <!-- download CSV included by default -->
+                <export has-values="recordValues.length>0" reference="reference" style="position:relative;" disabled="!displayReady"></export>
```
I think `has-values` must always be true for record app. We always want to show the button. 

`recordValues` might be empty (although that's not a normal case and it can only happen when the visible-columns is defined as an empty list) even though the data exists. Also we're going to show an error if the record doesn't exist so showing the button won't be harmful.

jrchudy commented:
@RFSH I changed the value for that directive to rely on `displayReady`. I also changed the name of the directive property from `hasValues` to `allowExport`.
